File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        frequency_table = defaultdict(int)

        for num in nums:
            frequency_table[num] += 1

        result = []

        sorted_values = sorted(frequency_table.values(), reverse = True)[:k]

        for key, value in frequency_table.items():
            if value in sorted_values:
                result.append(key)

        return result

        

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> elements(nums.begin(), nums.end());
        return (nums.size() == elements.size()) ? false : true;
    }
};